src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,501 @@
+        note_onsets_ready[number] = current_idx
+    return notes
+
+
+class Pop2PianoTokenizer(PreTrainedTokenizer):
+    """
+    Constructs a Pop2Piano tokenizer. This tokenizer does not require training.
+
+    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer
+    to: this superclass for more information regarding those methods. However the code does not allow that and only
+    supports composing from various genres.
+
+    Args:
+        vocab_file (`str`):
+            Path to the tokenizer file which contains token-ids such as `TOKEN_SPECIAL`, `DEFAULT_VELOCITY`.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+
+    def __init__(
+        self,
+        vocab_file,
+        unk_token="-1",
+        eos_token="1",
+        pad_token="0",
+        bos_token="2",
+        **kwargs,
+    ):
+        unk_token = AddedToken(unk_token, lstrip=False, rstrip=False) if isinstance(unk_token, str) else unk_token
+        eos_token = AddedToken(eos_token, lstrip=False, rstrip=False) if isinstance(eos_token, str) else eos_token
+        pad_token = AddedToken(pad_token, lstrip=False, rstrip=False) if isinstance(pad_token, str) else pad_token
+        bos_token = AddedToken(bos_token, lstrip=False, rstrip=False) if isinstance(bos_token, str) else bos_token
+
+        super().__init__(
+            unk_token=unk_token,
+            eos_token=eos_token,
+            pad_token=pad_token,
+            bos_token=bos_token,
+            **kwargs,
+        )
+
+        with open(vocab_file, "rb") as t_file:
+            self.encoder = json.load(t_file)
+
+        self.vocab_size_special = self.encoder["vocab_size_special"]
+        self.vocab_size_note = self.encoder["vocab_size_note"]
+        self.vocab_size_velocity = self.encoder["vocab_size_velocity"]
+        self.vocab_size_time = self.encoder["vocab_size_time"]
+        self.num_bars = self.encoder["num_bars"]
+
+    @property
+    def vocab_size(self):
+        """Returns the vocabulary size of the tokenizer."""
+        return self.vocab_size_special + self.vocab_size_note + self.vocab_size_time + self.vocab_size_velocity
+
+    def get_vocab(self):
+        """Returns the vocabulary of the tokenizer."""
+        return self.encoder
+
+    # copied from the official pop2piano implementation with little modification
+    # Please see https://github.com/sweetcocoa/pop2piano/blob/fac11e8dcfc73487513f4588e8d0c22a22f2fdc5/midi_tokenizer.py#L48
+    def _convert_id_to_token(self, token: int, time_idx_offset: int):
+        """
+        Decodes the tokens generated by the transformer.
+
+        Args:
+            token (`int`):
+                This denotes the token ids generated by the transformers to be converted to Midi tokens.
+            time_idx_offset (`int`):
+                This is a parameter which is used for TIME TOKEN when converting to Midi.
+        """
```

Comment:
return type should be included in the signature and here

Reply:
done